mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,51 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last and 'last_ckpt' in runner.message_hub.runtime_info:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            self._publish_model(runner, last_ckpt)
+
+        if getattr(self, 'best_ckpt_path', None) is not None:
+            self._publish_model(runner, self.best_ckpt_path)
+        if getattr(self, 'best_ckpt_path_dict', None) is not None:
+            for key, best_ckpt in self.best_ckpt_path_dict.items():
+                self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, out_file: str) -> None:
+        """Publish the checkpoint.
+
+        Args:
+            runner (Runner): The runner of the training process.
+            out_file (str): The checkpoint path that ought to be published.
+        """
+        checkpoint = _load_checkpoint(out_file)
+        published_keys = self.published_keys
+        removed_keys = []
+        for key in list(checkpoint.keys()):
+            if key not in published_keys:
+                removed_keys.append(key)
+                checkpoint.pop(key)
+        if removed_keys:
+            print_log(
+                f'Key {removed_keys} will be removed because they are not '
+                'found in published_keys. If you want to keep them, '
+                f'please set `{removed_keys}` in published_keys',
+                logger='current')
+        final_file = osp.splitext(
+            out_file)[0] + f'-published-{runner.timestamp}.pth'
+        save_checkpoint(checkpoint, final_file)
+        print_log(
+            f'The published model is saved at {final_file}.', logger='current')
```

Comment:
```suggestion
    def _publish_model(self, runner, ckpt_path: str) -> None:
        """Remove unnecessary keys from ckpt_path and save the new checkpoint.

        Args:
            runner (Runner): The runner of the training process.
            ckpt_path (str): The checkpoint path that ought to be published.
        """
        checkpoint = _load_checkpoint(ckpt_path)
        published_keys = self.published_keys
        removed_keys = []
        for key in list(checkpoint.keys()):
            if key not in published_keys:
                removed_keys.append(key)
                checkpoint.pop(key)
        if removed_keys:
            print_log(
                f'Key {removed_keys} will be removed because they are not '
                'found in published_keys. If you want to keep them, '
                f'please set `{removed_keys}` in published_keys',
                logger='current')
        final_path = osp.splitext(
            ckpt_path)[0] + f'-published-{runner.timestamp}.pth'
        save_checkpoint(checkpoint, final_path)
        print_log(
            f'The published model is saved at {final_path}.', logger='current')
```